aci/resource_aci_bfdmhnodepol.go

```diff
@@ -0,0 +1,252 @@
+	if err != nil {
+		return nil, err
+	}
+	bfdMhNodePol := models.BFDMultihopNodePolicyFromContainer(bfdMhNodePolCont)
+	if bfdMhNodePol.DistinguishedName == "" {
+		return nil, fmt.Errorf("BFD Multihop Node Policy %s not found", dn)
+	}
+	return bfdMhNodePol, nil
+}
+
+func setBFDMultihopNodePolicyAttributes(bfdMhNodePol *models.BFDMultihopNodePolicy, d *schema.ResourceData) (*schema.ResourceData, error) {
+	d.SetId(bfdMhNodePol.DistinguishedName)
+	d.Set("description", bfdMhNodePol.Description)
+	bfdMhNodePolMap, err := bfdMhNodePol.ToMap()
+	if err != nil {
+		return d, err
+	}
+	dn := d.Id()
+	if dn != bfdMhNodePol.DistinguishedName {
+		d.Set("tenant_dn", "")
+	} else {
+		d.Set("tenant_dn", GetParentDn(bfdMhNodePol.DistinguishedName, fmt.Sprintf("/"+models.RnBfdMhNodePol, bfdMhNodePolMap["name"])))
+	}
+	d.Set("admin_state", bfdMhNodePolMap["adminSt"])
+	d.Set("annotation", bfdMhNodePolMap["annotation"])
+	d.Set("detection_multiplier", bfdMhNodePolMap["detectMult"])
+	d.Set("min_rx_interval", bfdMhNodePolMap["minRxIntvl"])
+	d.Set("min_tx_interval", bfdMhNodePolMap["minTxIntvl"])
+	d.Set("name", bfdMhNodePolMap["name"])
+	d.Set("name_alias", bfdMhNodePolMap["nameAlias"])
+	return d, nil
+}
+
+func resourceAciBFDMultihopNodePolicyImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+	bfdMhNodePol, err := getRemoteBFDMultihopNodePolicy(aciClient, dn)
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled, err := setBFDMultihopNodePolicyAttributes(bfdMhNodePol, d)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciBFDMultihopNodePolicyCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] BFD Multihop Node Policy: Beginning Creation")
+	aciClient := m.(*client.Client)
+	desc := d.Get("description").(string)
+	name := d.Get("name").(string)
+	TenantDn := d.Get("tenant_dn").(string)
+
+	bfdMhNodePolAttr := models.BFDMultihopNodePolicyAttributes{}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		bfdMhNodePolAttr.Annotation = Annotation.(string)
+	} else {
+		bfdMhNodePolAttr.Annotation = "{}"
+	}
+
+	if AdminSt, ok := d.GetOk("admin_state"); ok {
+		bfdMhNodePolAttr.AdminSt = AdminSt.(string)
+	}
+
+	if DetectMult, ok := d.GetOk("detection_multiplier"); ok {
+		bfdMhNodePolAttr.DetectMult = DetectMult.(string)
+	}
+
+	if MinRxIntvl, ok := d.GetOk("min_rx_interval"); ok {
+		bfdMhNodePolAttr.MinRxIntvl = MinRxIntvl.(string)
+	}
+
+	if MinTxIntvl, ok := d.GetOk("min_tx_interval"); ok {
+		bfdMhNodePolAttr.MinTxIntvl = MinTxIntvl.(string)
+	}
+
+	if Name, ok := d.GetOk("name"); ok {
+		bfdMhNodePolAttr.Name = Name.(string)
+	}
+
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		bfdMhNodePolAttr.NameAlias = NameAlias.(string)
+	}
```

Comment:
should we create function for this since exact block is used in update?

Reply:
@shrsr @sajagana @lhercot what is your take on this?

Reply:
Yeah, we have done that in a few resources.

@gmicol for your reference:
https://github.com/CiscoDevNet/terraform-provider-aci/pull/1063/files#diff-33750ff9cf6ad392d7ab500d19e4155b5b337bdfc1de936f03c039e31ffbd433